src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
+                        }
+                        else
+                        {
+                            _logger.LogError($"No geocoding results for address \"{address}\"");
+                            TempData["AlertDanger"] = $"Unable to locate address <strong>\"{address}\"</strong>.";
+                        }
+                    }
+                    catch (HttpRequestException ex)
+                    {
+                        _logger.LogCritical(ex, $"Google API error: {ex.Message}");
+                        TempData["AlertDanger"] = "An error occured, please try again later.";
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogCritical(ex, ex.Message);
+                        TempData["AlertDanger"] = "An error occured, please try again later.";
+                    }
+                }
+            }
+            if (mobile == 1)
+            {
+                return View(new Location { ShowLocation = true });
+            }
+            return View("Locations");
+        }
+
+        [HttpGet("Locations")]
+        [HttpGet("Locations/[action]")]
+        public async Task<IActionResult> Find(double latitude = 0, double longitude = 0, string address = null, int? mobile = null)
+        {
+            Location[] locations = {};
+
+            if((latitude!=0 && longitude!=0) && address==null)
+            {
+                var viewModel = LibraryLookup(latitude, longitude);
+                viewModel.ShowLocation = true;
+                var latlng = $"{latitude},{longitude}";
+
+                // try to get the zip code to display to the user
+                try
+                {
+                    using (var client = new HttpClient())
+                    {
+                        var apikey = _config["APIKey"];
+
+                        var response = await client.GetAsync($"https://maps.googleapis.com/maps/api/geocode/json?latlng={latlng}&key={apikey}");
+                        response.EnsureSuccessStatusCode();
+
+                        GeocodeResult geoResult = null;
+                        var stringResult = await response.Content.ReadAsStringAsync();
+
+                        try
+                        {
+                            geoResult = JsonConvert.DeserializeObject<GeocodeResult>(stringResult);
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogError($"Error parsing Geocode API JSON: {ex.Message} - {stringResult}");
+                        }
+
+                        if (geoResult?.Results?.Count() > 0)
+                        {
+                            viewModel.Address = geoResult.Results?
+                                .FirstOrDefault()?
+                                .AddressComponents?
+                                .Where(_ => _.Types.Any(t => t == "postal_code"))
```

Comment:
Use `_` in place of `t` here. Also, `Where` is not necessary, place the where clause in the `.FirstOrDefault()` method.